### DISCUSSION TITLE
What's way work with Mapped is better?

### DISCUSSION THREAD

**Discussion by oleg-korshunov** [sqlalchemy/sqlalchemy#9649] [Usage Questions] [answered]
Hi guys, I started using syntaxys SQLAlchemy 2.0 and in docs exist example how to work with relationship in the one file, I have the follow question, for example I have the two table
```python
# author.py
from database import Base
from typing import List
from sqlalchemy import Integer, String
from sqlalchemy.orm import Mapped, Relationship
from sqlalchemy.orm import mapped_column

class Author(Base):
    __tablename__ = "author"
    id = mapped_column(Integer, primary_key=True)
    name = mapped_column(String(50), nullable=False)
    # one-to-many
    book: Mapped[List["Book"]] = Relationship(back_populates="author")
```
```python
from database import Base
from sqlalchemy import Integer, String, ForeignKey
from sqlalchemy.orm import Mapped, Relationship
from sqlalchemy.orm import mapped_column
# book.py
class Book(Base):
    __tablename__ = "book"
    id = mapped_column(Integer, primary_key=True)
    name = mapped_column(String(50), nullable=False)
    author_id = mapped_column(Integer, ForeignKey("author.id"))
    author: Mapped["Author"] = Relationship(back_populates="book")
```
and If I want write some query in repositry.py for Author table, I need to import both tables

```python
from author import Author
from book import Book
from database import Base, engine
from sqlalchemy.orm import sessionmaker
from sqlalchemy import select
Base.metadata.create_all(engine)
Ssession = sessionmaker(bind=engine)
session = Ssession()
query = select(Author).where(Author.name == "Dean")
print(session.execute(query).all())
```
Is exist way with Mapped syntax how to don't import tables from relationship?

**Comment by zzzeek** [comment, marked as answer]
when the Python interpreter runs , the code that you've written that defines the Author class and the Book class will only be known to the interpreter if you import those modules using the `import` statement.   So there is no alternative to not `import` ing both modules.   The question you refer to at #9522 has to do with if the `import` statement needs to be **in the same file as the relationship()**, and for that question, the answer is no.  However, when you **use** the mappings, both files have to have been read by the Python interpreter.

the usual way this is achieved is by having a package called `models`, and within the `models/__init__.py` file, both `author` and `book` will have been imported.  However `book` and `author` themselves do not need to refer to each other, which otherwise would create the circular import problem discussed in #9522.

**Reply by oleg-korshunov** [reply]
Thanks for the answer! In my project I added all models in package models/\_\_init__.py, and import it in main.py and now everything works perfectly!
